atd-etl/cris_import/cris_import.py

```diff
@@ -225,45 +270,107 @@ def specify_extract_location():
+        )
+        ssh_tunnel.start()
+
+        pg = psycopg2.connect(
+            host="localhost",
+            port=ssh_tunnel.local_bind_port,
+            user=DB_USER,
+            password=DB_PASS,
+            dbname=DB_NAME,
+            sslmode=DB_SSL_REQUIREMENT,
+            sslrootcert="/root/rds-combined-ca-bundle.pem",
+        )
+
+        uploads_prefix = f"{DEPLOYMENT_ENVIRONMENT}/inbox/"
+        # Get list of all objects in the bucket with the specified prefix
+        objects = s3.list_objects(Bucket=bucket, Prefix=uploads_prefix)
+
+        cursor = pg.cursor()
 
-        zip_tmpdir = tempfile.mkdtemp()
-        rsync = None
-        try:
-            rsync = sysrsync.run(
-                verbose=True,
-                options=["-a"],
-                source_ssh=SFTP_ENDPOINT,
-                source="/home/txdot/*zip",
-                sync_source_contents=False,
-                destination=zip_tmpdir,
-                private_key=key_directory + "/id_ed25519",
-                strict_host_key_checking=False,
+        for obj in objects["Contents"]:
+            full_object_path = obj["Key"]
+            object_name = os.path.basename(full_object_path)
+
+            # Skip the "folder" object
+            if full_object_path == uploads_prefix:
+                continue
+
+            # Extract the S3 prefix from the full object path
+            object_path = os.path.dirname(full_object_path)
+
+            # Check if the object already exists in the database
+            cursor.execute(
+                "SELECT * FROM cris_import_log WHERE object_path = %s AND object_name = %s",
+                (object_path, object_name),
             )
-        except:
-            print("No files to copy..")
-            # we're really kinda out of work here, so we're going to bail
-            quit()
-        print("Rsync return code: " + str(rsync.returncode))
-        # check for a OS level return code of anything non-zero, which
-        # would indicate to us that the child proc we kicked off didn't
-        # complete successfully.
-        # see: https://www.gnu.org/software/libc/manual/html_node/Exit-Status.html
-        if rsync.returncode != 0:
-            return False
-        print("Temp Directory: " + zip_tmpdir)
-        return zip_tmpdir
+            result = cursor.fetchone()
+
+            # If the object does not exist, insert a new record
+            if result is None:
+                cursor.execute(
+                    """
+                    INSERT INTO cris_import_log (object_path, object_name, first_seen)
+                    VALUES (%s, %s, %s)
+                    """,
+                    (object_path, object_name, datetime.datetime.utcnow()),
+                )
+
+        # Query all uploads where import_attempted = 0
+        cursor.execute("SELECT * FROM cris_import_log WHERE import_attempted is false")
+        uploads_to_import = cursor.fetchall()
```

Comment:
instead of `uploads_to_import` — is `extracts_to_process` (or similar) a more accurate var name?

Reply:
💯 Yes -- this naming made a lot more sense with my old S3 structure, and it's nonsensical with the enhancements we've done. This symbol is renamed. https://github.com/cityofaustin/atd-vz-data/pull/1445/commits/2baf51c249f8c2b751f60f99a32583f2aa4eb1a3